### PR TITLE
Add 0.10.0 changelog

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:
It's a changelog!

**Special notes for your reviewer**:
- Draft still pending release date and Ingress v1 PR info, but the rest should all be ready. Probably has some minor omissions since the change history is pretty big and not everything was attached to the milestone, but should hit all the major points.
- Didn't cover Knative now "requiring" class information, since I'm fairly sure this wasn't really possible before (I don't think Knative allows you to create them, and discussion with the Knative team appears to confirm this), it was just something we supported in the code.

### CODE REVIEW

**Comment by mflendrich** [comment]
> Didn't cover Knative now "requiring" class information, since I'm fairly sure this wasn't really possible before (I don't think Knative allows you to create them, and discussion with the Knative team appears to confirm this), it was just something we supported in the code.

Should this be understood as
    a) this changes the interface of KIC to users/other systems but I think it's unlikely that someone will hit this, or
    b) this changes an internal detail of KIC that is theoretically impossible to see on KIC's interface, either documented or undocumented

?

If it's a), LGTM.
If it's b), let's not omit that in the changelog because https://www.hyrumslaw.com/ .

**Review by mflendrich — CHANGES_REQUESTED** [review]
ditto

**Comment by rainest** [comment]
What was the ditto? It looks like that was intended as a line comment.

Re Knative class, wasn't really sure--per their devs, it's something they really don't expect and don't intend to support, and Knative itself is new enough that such a loose stipulation is enough to push it into "eh, it's quite unlikely that anyone does rely on this", but went and checked since I was in the area.

In practice, our code _would_ accept classless Knative Ingresses; that was a side effect of using the plain Ingress matching behavior for everything. To get one, however, you'd need to write your own Knative implementation or create Knative Ingresses by hand (I think this is possible, but it kinda defeats the purpose of using Knative to begin with, and it'd probably break a bunch of other stuff).

The standard Knative implementation applies a `"networking.knative.dev/ingress.class": "istio.ingress.networking.knative.dev"` annotation if none is provided in the `knative-serving/config-network` ConfigMap: https://github.com/knative/serving/blob/f141b23104f37206005a3304d4d9951ebfa335e0/vendor/knative.dev/networking/pkg/network.go#L250

It's not wholly impossible if we consider our code in isolation, but outside really contrived scenarios it's impossible in practice.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Table of Contents
 
+ - [0.10.0](#0100---someday)
```
Comment (hbagdi) [inline CHANGELOG.md:3]: 09/15

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +23,100 @@
...
+  default. Kong recommends adding this annotation to Ingresses that previously
+  did not have it, but you can override this change and instruct the controller
+  to process Ingresses without this annotation if desired. See the [ingress
+  class documentation](https://example.com/link-tbd) for details.
+  [#767](https://github.com/Kong/kubernetes-ingress-controller/pull/767)
+- KongConsumer resources now require `kubernetes.io/ingress.class` annotations
+  by default. This change can also be overriden using a flag.
+  [#767](https://github.com/Kong/kubernetes-ingress-controller/pull/767)
+- TCPIngress resources now require `kubernetes.io/ingress.class` annotations.
+  This change _cannot_ be overriden.
```
Comment (hbagdi) [inline CHANGELOG.md:40]: nit, not a blocker: Instead use "This is not configurable."

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +23,100 @@
...
+  [#767](https://github.com/Kong/kubernetes-ingress-controller/pull/767)
+- CA certificate secrets now require `kubernetes.io/ingress.class` annotations.
+  This change _cannot_ be overriden.
```
Comment (hbagdi) [inline CHANGELOG.md:43]: nit, not a blocker: Instead use "This is not configurable."

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +23,100 @@
...
+  will also log a warning if it finds any global KongPlugins that are still in
+  place.
+  [#751](https://github.com/Kong/kubernetes-ingress-controller/pull/751)
+
+
+#### Added
+
+- Added support for [Ingress
+  v1](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1453-ingress-api#summary-of-the-proposed-changes).
+  [#TBD](https://github.com/Kong/kubernetes-ingress-controller/pull/TBD).
```
Comment (hbagdi) [inline CHANGELOG.md:58]: Please instead point to the umbrella issue: https://github.com/Kong/kubernetes-ingress-controller/issues/590

File: CHANGELOG.md [hunk 5]
```diff
@@ -22,6 +23,100 @@
...
+- Added support for the port mapping functionality in Kong versions 2.1 and
+  newer in example manifests. This feature [improves Kong's functionality when
+  behind a load balancer that uses different ports than Kong's proxy
+  listens](https://github.com/Kong/kong/pull/5861).
+  [#753](https://github.com/Kong/kubernetes-ingress-controller/pull/753)
+- Added support for the `ingress.kubernetes.io/force-ssl-redirect` annotation.
+  [#745](https://github.com/Kong/kubernetes-ingress-controller/pull/745)
+- Transitioned to structured logging.
+  [#748](https://github.com/Kong/kubernetes-ingress-controller/pull/748)
+- Added flags to enable processing of Ingress and KongConsumer resources
```
Comment (hbagdi) [inline CHANGELOG.md:68]: I'll highly recommend to organize this is a little differently.
Instead, have a bullet point on "Following new flags have been introduced" and then have a sub-point to document each of the flag that has been introduced in this release.
This helps with having a single document which can tell us when was which configuration and feature introduced.

**Comment by hbagdi** [comment]
Also please change this to a regular PR from draft.

**Review by mflendrich — APPROVED** [review]

**Review by mflendrich — APPROVED** [review]

**Review by mflendrich — APPROVED** [review]

**Review by hbagdi — APPROVED** [review]

**Pull request merged**